lib/recursive-methods.rb

```diff
@@ -1,49 +1,108 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: 0(n)
+# Space complexity: 0(n)
+# source: video example in class
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+    if n < 0
+        raise ArgumentError
+    elsif n == 1 || n == 0
+        return 1 
+    else
+        return n * factorial(n-1)
+    end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: 0(n)
+# Space complexity: 0(n)
 def reverse(s)
-    raise NotImplementedError, "Method not implemented"
+    if s.length <= 1
+        return s 
+    else
+        return s[-1] + reverse(s[1...-1]) + s[0]
```

Comment:
Since the `s[1...-1]` creates a new array with each recursive call, your space complexity is going to be O(n^2) and I believe  since this copies the array, it also affects the time complexity.